knowhere/index/vector_index/IndexHNSW.cpp

```diff
@@ -231,6 +238,15 @@ IndexHNSW::Query(const DatasetPtr& dataset_ptr, const Config& config, const fais
     // LOG_KNOWHERE_DEBUG_ << "IndexHNSW::Query finished, show statistics:";
     // LOG_KNOWHERE_DEBUG_ << GetStatistics()->ToString();
 
+    // set visit_info json string into result dataset
+    if (feder_result != nullptr) {
+        Config json_visit_info, json_id_set;
+        nlohmann::to_json(json_visit_info, feder_result->visit_info_);
+        nlohmann::to_json(json_id_set, feder_result->id_set_);
+        delete feder_result;
```

Comment:
Suggest to use `unique_ptr` here. For `searchKnn` we can pass `const unique_ptr<>&`. This way we can avoid the raw pointer and new/delete

Reply:
updated